spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -258,6 +295,125 @@ protected abstract MessageProducer createConsumerEndpoint(ConsumerDestination de
 	protected void afterUnbindConsumer(ConsumerDestination destination, String group, C consumerProperties) {
 	}
 
+	/**
+	 * Build an errorChannelRecoverer that writes to a pub/sub channel for the destination.
+	 * @param destination the destination.
+	 * @param extendedConsumerProperties the properties.
+	 * @param errorMessageStrategy
+	 * @return the recoverer.
+	 */
+	protected final ErrorInfrastructure registerErrorInfrastructure(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<?> extendedConsumerProperties) {
+		ErrorMessageStrategy errorMessageStrategy = getErrorMessageStrategy();
+		ConfigurableListableBeanFactory beanFactory = getApplicationContext().getBeanFactory();
+		String errorChannelName = errorsBaseName(destination, group, extendedConsumerProperties);
+		SubscribableChannel errorChannel;
+		if (getApplicationContext().containsBean(errorChannelName)) {
+			errorChannel = getApplicationContext().getBean(errorChannelName, SubscribableChannel.class);
+		}
+		else {
+			errorChannel = new ErrorChannel();
+			beanFactory.registerSingleton(errorChannelName, errorChannel);
+			errorChannel = (SubscribableChannel) beanFactory.initializeBean(errorChannel, errorChannelName);
+		}
+		ErrorMessageSendingRecoverer recoverer;
+		if (errorMessageStrategy == null) {
+			recoverer = new ErrorMessageSendingRecoverer(errorChannel);
+		}
+		else {
+			recoverer = new ErrorMessageSendingRecoverer(errorChannel, errorMessageStrategy);
+		}
+		String recovererBeanName = getErrorRecovererName(destination, group, extendedConsumerProperties);
+		beanFactory.registerSingleton(recovererBeanName, recoverer);
+		beanFactory.initializeBean(recoverer, recovererBeanName);
+		MessageHandler handler = getErrorMessageHandler(destination, group, extendedConsumerProperties);
+		if (handler == null) {
+			handler = getDefaultErrorMessageHandler(errorChannel);
+		}
+		String errorMessageHandlerName = getErrorMessageHandlerName(destination, group, extendedConsumerProperties);
+		beanFactory.registerSingleton(errorMessageHandlerName, handler);
+		beanFactory.initializeBean(handler, errorMessageHandlerName);
+		errorChannel.subscribe(handler);
+		return new ErrorInfrastructure(errorChannel, recoverer, handler);
```

Comment:
I'm not sure yet why I would need such a complex object in return, but  believe some JavaDocs on the matter will be helpful for any target `Binder` implementators.
Thanks